packages/trader/src/AppV2/Components/MarketCategoryItem/market-category-item.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect, forwardRef, Ref } from 'react';
+import { Tag, Text, useSnackbar } from '@deriv-com/quill-ui';
+import { StandaloneStarFillIcon, StandaloneStarRegularIcon } from '@deriv/quill-icons';
+import { Localize } from '@deriv/translations';
+import SymbolIconsMapper from '../SymbolIconsMapper/symbol-icons-mapper';
+import { ActiveSymbols } from '@deriv/api-types';
+import clsx from 'clsx';
+import { observer, useStore } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { useModulesStore } from 'Stores/useModulesStores';
+
+type TMarketCategoryItem = {
+    item: ActiveSymbols[0];
+    selectedSymbol: string;
+    setSelectedSymbol: (input: string) => void;
+    setIsOpen: (input: boolean) => void;
+};
+
+const MarketCategoryItem = forwardRef(
+    ({ item, selectedSymbol, setSelectedSymbol, setIsOpen }: TMarketCategoryItem, ref: Ref<HTMLDivElement>) => {
+        const [isFavorite, setIsFavorite] = useState(false);
+        const { onChange: onSymbolChange } = useTraderStore();
+        const { markets } = useModulesStore();
+        const { favoriteSymbols, setFavoriteSymbols, removeFavoriteSymbol } = markets;
+        const { addSnackbar } = useSnackbar();
+        const { ui } = useStore();
+        const { is_dark_mode_on } = ui;
+
+        useEffect(() => {
+            setIsFavorite(favoriteSymbols.includes(item.symbol));
+        }, [favoriteSymbols, item.symbol]);
+
+        const handleSelect = async (e: React.MouseEvent<HTMLSpanElement>) => {
+            const symbol = (e.target as HTMLSpanElement).getAttribute('data-symbol');
```

Comment:
Is there any possibility to avoid using `data-symbol`? If `data-symbol` is `==={item.symbol}`, may we use `item.symbol` directly in this function?
Please, ignore if irrelevant

Reply:
doesnt seem to be working well with symbol switching